pkg/dockerregistry/server/pullthroughblobstore.go

```diff
@@ -104,10 +107,40 @@ func (pbs *pullthroughBlobStore) Get(ctx context.Context, dgst digest.Digest) ([
 // setResponseHeaders sets the appropriate content serving headers
 func setResponseHeaders(w http.ResponseWriter, length int64, mediaType string, digest digest.Digest) {
 	w.Header().Set("Content-Type", mediaType)
+	w.Header().Set("Content-Length", fmt.Sprintf("%d", length))
```

Comment:
Let `http.ServeContent` set this. The value will be shorter than `length` for range requests. Set it only in case `http.ServeContent` doesn't run.